Short-Answer/Algorithms_Answers.md

```diff
@@ -1,15 +1,50 @@
-#### Please add your answers to the ***Analysis of  Algorithms*** exercises here.
-
-## Exercise I
-
-a)
-
-
-b)
-
-
-c)
-
-## Exercise II
-
-
+#### Please add your answers to the ***Analysis of  Algorithms*** exercises here.
+
+## Exercise I
+
+a)the while loop is dependent on the increase of n. linear O(n)
+
+b) we will be looping over n and while j is less than n we do the arithemetic problems which will repeatedly increase j    O(log n)
+```
+b)  sum = 0
+    for i in range(n): Since it's a loop it gets linear O(n)
+      j = 1
+      while j < n: 
+        j *= 2

```

Comment:
It looks like the final answer you provided was `O(log n)`?

Your analysis of the inner while-loop is generally correct, you just need to point out that that `j` doubles in size while `j < n`. The halving is why you get `O(log n)` for the inner while-loop.

Then if you account for the outer for-loop which runs at `O(n)` to produce a final result of `O(n * log n)`

I'm counting this correct though because you note the outer for-loop is `O(n)` in your comments.

Nice job with the rest of these!